pkg/controller/migmigration/dvm.go

```diff
@@ -0,0 +1,163 @@
+package migmigration
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	migapi "github.com/konveyor/mig-controller/pkg/apis/migration/v1alpha1"
+	dvmc "github.com/konveyor/mig-controller/pkg/controller/directvolumemigration"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8sclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func (t *Task) createDirectVolumeMigration() error {
+	existingDvm, err := t.getDirectVolumeMigration()
+	if err != nil {
+		return err
+	}
+	// If already created exit with no error
+	if existingDvm != nil {
+		return nil
+	}
+	dvm := t.buildDirectVolumeMigration()
+	if dvm == nil {
+		return errors.New("failed to build directvolumeclaim list")
```

Comment:
I believe this should be an error that should surface to the user only because it means something went catastrophically wrong. This would mean that the controller has indicated we are supposed to be creating a DVM but building up the resource returned nothing. that's my take at least

Reply:
@dymurray since it is an error state that we want to bubble up to user, should we set a `Warn` Condition here?

also @djwhatle do you mind if we take this as follow on since there needs to be some discussion around how to handle this?

Reply:
@alaypatel07 Yeah -- see the other discussion with @dymurray, but having no dvm probably shouldn't be an error or a warning, since it just means there weren't any filesystem copy PVCs. If the user chooses move or snapshot copy for everything, we won't create a dvm, and that's fine. Likewise the WaitFor... case won't find a DVM so it won't have to wait -- but I think we can defer that fix (and removing this error) to a follow-on PR.